spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/config/KinesisBinderConfiguration.java

```diff
@@ -59,8 +60,13 @@ public AmazonKinesisAsync amazonKinesis(
 	}
 
 	@Bean
-	public KinesisStreamProvisioner provisioningProvider(AmazonKinesisAsync amazonKinesis) {
-		return new KinesisStreamProvisioner(amazonKinesis, this.configurationProperties);
+	public KinesisStreamHandler kinesisStreamHandler(AmazonKinesisAsync amazonKinesis) {
+		return new KinesisStreamHandler(amazonKinesis);
```

Comment:
I wouldn't introduce extra level if we can just stick with the existing clear API from the `AmazonKinesis`.